src/site/content/en/blog/bluetooth/index.md

```diff
@@ -0,0 +1,549 @@
+  </video>
+  <figcaption class="w-figcaption">Bluetooth device user prompt.</figcaption>
+</figure>
+
+
+The `navigator.bluetooth.requestDevice` function takes a mandatory Object that
+defines filters. These filters are used to return only devices that match some
+advertised Bluetooth GATT services and/or the device name.
+
+For instance, requesting Bluetooth devices advertising the [Bluetooth GATT
+Battery Service] is this simple:
+
+```js
+navigator.bluetooth.requestDevice({ filters: [{ services: ['battery_service'] }] })
+.then(device => { /* ... */ })
+.catch(error => { console.log(error); });
+```
+
+If your Bluetooth GATT Service is not on the list of [the standardized Bluetooth
+GATT services] though, you may provide either the full Bluetooth UUID or a short
+16- or 32-bit form.
+
+```js
+navigator.bluetooth.requestDevice({
+  filters: [{
+    services: [0x1234, 0x12345678, '99999999-0000-1000-8000-00805f9b34fb']
+  }]
+})
+.then(device => { /* ... */ })
+.catch(error => { console.log(error); });
+```
+
+You can also request Bluetooth devices based on the device name being advertised
+with the `name` filters key, or even a prefix of this name with the `namePrefix`
+filters key. Note that in this case, you will also need to define the
+`optionalServices` key to be able to access some services. If you don't, you'll
+get an error later when trying to access them.
+
+```js
+navigator.bluetooth.requestDevice({
+  filters: [{
+    name: 'Francois robot'
+  }],
+  optionalServices: ['battery_service']
+})
+.then(device => { /* ... */ })
+.catch(error => { console.log(error); });
+```
+
+Finally, instead of `filters` you can use the `acceptAllDevices` key to show all
+nearby Bluetooth devices. You will also need to define the `optionalServices`
+key to be able to access some services. If you don't, you'll get an error later
+when trying to access them.
+
+```js
+navigator.bluetooth.requestDevice({
+  acceptAllDevices: true,
+  optionalServices: ['battery_service']
+})
+.then(device => { /* ... */ })
+.catch(error => { console.log(error); });
+```
+
+{% Aside 'caution' %}
+This may result in a bunch of unrelated devices being shown in the chooser and
+energy being wasted as there are no filters. Use it with caution.
+{% endAside %}
+
+### Connect to a Bluetooth Device
+
+So what do you do now that you have a `BluetoothDevice` returned from
+`navigator.bluetooth.requestDevice`'s Promise? Let's connect to the Bluetooth
+remote GATT Server which holds the service and characteristic definitions.
+
+```js
+navigator.bluetooth.requestDevice({ filters: [{ services: ['battery_service'] }] })
+.then(device => {
+  // Human-readable name of the device.
+  console.log(device.name);
+
+  // Attempts to connect to remote GATT Server.
+  return device.gatt.connect();
+})
+.then(server => { /* ... */ })
+.catch(error => { console.log(error); });
+```
+
+### Read a Bluetooth Characteristic
```

Comment:
Note to self: In this case it seems like `Bluetooth Characteristic` should be capitalized